### PR TITLE
feat(CDAP-20496): Add READ_REPOSITORY, WRITE_REPOSITORY, UPDATE_REPOSITORY_METADATA permissions in cdap proto

### PR DESCRIPTION
**Pull request by de-lan**
## What:

Add READ_REPOSITORY, WRITE_REPOSITORY, UPDATE_REPOSITORY_METADATA permissions in cdap proto

### CODE REVIEW

File: cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto.security;
+
+/**
+ * Permissions specifically  for CDAP Namespace.
```
Comment (dli357) [inline cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java]: nit: extra space in the comment

File: cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ */
+public enum NamespacePermission implements Permission {
+  /**
+   * Permission to read data in the repository of a namespace.
```
Comment (dli357) [inline cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java]: nit: Please add that these permissions are specifically used for SCM, e.g. `read data in the SCM repository`, same for other docs, thanks!

**Review by dli357 — APPROVED** [review]
2 nit comments, otherwise LGTM. Thanks!

File: cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java (hunk 1 above)
Comment (de-lan) [reply cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java]: nice catch! removed the extra space

File: cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java (hunk 2 above)
Comment (de-lan) [reply cdap-proto/src/main/java/io/cdap/cdap/proto/security/NamespacePermission.java]: Added SCM in comments, thanks!

**Pull request merged**